CleverRaven/Cataclysm-DDA
# Use mutation visibility as condition

#### Summary
none


#### Purpose of change
Address #67152


#### Describe the solution
As there are visible mutations displayed in the NPC's info window, you can use them as condition for dialogue.
![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/22912139/92911649-b822-42e1-8bd9-956bc3697c28)

Also make the arsonist dialogue use this new condition to determine the responses.

Hopefully this has the prospects to improve immersion and enhance role playing.



#### Describe alternatives you've considered
Allow using mutation visibility values in eoc context for more customized judgement.


#### Testing
1. Go to the arsonist, find that I can see they have feline ears, and the corresponding response shows up.
2. Set my character's per to 0, now their feline ears trait doesn't show up in the NPC info window, and the corresponding response doesn't show up.


#### Additional context
1. Should I add a `sees()` check to this?
2. I believe more conversation can be ported to use this instead of viewing from God's perspective and use `u_has_trait`.
3. Should I imitate `u_has_any_trait` and implement something like `u_has_any_visible_trait`?
